src/pgwire/src/codec.rs

```diff
@@ -476,7 +506,220 @@ fn decode_terminate(mut _buf: Cursor) -> Result<FrontendMessage, io::Error> {
     Ok(FrontendMessage::Terminate)
 }
 
-fn decode_password(mut buf: Cursor) -> Result<FrontendMessage, io::Error> {
+fn decode_auth(mut buf: Cursor) -> Result<FrontendMessage, io::Error> {
+    let mut value = Vec::new();
+    while let Ok(b) = buf.read_byte() {
+        value.push(b);
+    }
+    Ok(FrontendMessage::RawAuthentication(value))
+}
+
+fn expect(buf: &mut Cursor, expected: &[u8]) -> Result<(), io::Error> {
+    for i in 0..expected.len() {
+        if buf.read_byte()? != expected[i] {
+            return Err(input_err(format!(
+                "Invalid SASL initial response: expected '{}'",
+                std::str::from_utf8(expected).unwrap_or("invalid UTF-8")
+            )));
+        }
+    }
+    Ok(())
+}
+
+fn read_until_comma(buf: &mut Cursor) -> Result<Vec<u8>, io::Error> {
+    let mut v = Vec::new();
+    while let Ok(b) = buf.peek_byte() {
+        if b == b',' {
+            break;
+        }
+        v.push(buf.read_byte()?);
+    }
+    Ok(v)
+}
+
+//   extensions = attr-val *("," attr-val)
```

Comment:
FYI: Noticed this comment came from https://github.com/postgres/postgres/blob/a48d1ef58652229521ba4b5070e19f857608b22e/src/backend/libpq/auth-scram.c#L933. Idk where the company stands on this but I would recommend doing a lil informal "citation" to give more context / information for future developers. We do this a bunch (i.e. `mz_acl_item.rs`)

Reply:
Added a comment above pointing to spec.